Mathlib/CategoryTheory/Grothendieck.lean

```diff
@@ -369,6 +417,84 @@ def pre (G : D ⥤ C) : Grothendieck (G ⋙ F) ⥤ Grothendieck F where
   map_id X := Grothendieck.ext _ _ (G.map_id _) (by simp)
   map_comp f g := Grothendieck.ext _ _ (G.map_comp _ _) (by simp)
 
+@[simp]
+theorem pre_id : pre F (𝟭 C) = 𝟭 _ := rfl
+
+/--
+An natural isomorphism between functors `G ≅ H` induces a natural isomorphism between the canonical
+morphism `pre F G` and `pre F H`, up to composition with
+`Grothendieck (G ⋙ F) ⥤ Grothendieck (H ⋙ F)`.
+-/
+def preNatIso {G H : D ⥤ C} (α : G ≅ H) :
+    pre F G ≅ (map (whiskerRight α.hom F)) ⋙ (pre F H) :=
+  NatIso.ofComponents
+    (fun X => (transportIso ⟨G.obj X.base, X.fiber⟩ (α.app X.base)).symm)
+    (fun f => by fapply Grothendieck.ext <;> simp [transport_hom])
+
+/--
+Given an equivalence of categories `G`, `preInv _ G` is the (weak) inverse of the `pre _ G.functor`.
+-/
+def preInv (G : D ≌ C) : Grothendieck F ⥤ Grothendieck (G.functor ⋙ F) := by
+  refine map ?_ ⋙ Grothendieck.pre (G.functor ⋙ F) G.inverse
+  rw [← Functor.assoc]
+  exact eqToHom (Functor.id_comp F) ≫ (whiskerRight G.counitInv F)
```

Comment:
```suggestion
def preInv (G : D ≌ C) : Grothendieck F ⥤ Grothendieck (G.functor ⋙ F) :=
  map (F.leftUnitor.inv ≫ whiskerRight G.counitInv F ≫ (Functor.associator _ _ _).hom) ⋙
    Grothendieck.pre (G.functor ⋙ F) G.inverse
```

Reply:
I removed `leftUnitor` and `associator` -- they are just `rfl` and somehow seem to break `preEquivalence`.